### DISCUSSION TITLE
Trying to build with C++ gives error with "WITH_IPOPT" flag

### DISCUSSION THREAD

**Discussion by PinyaColada** [casadi/casadi#3509] [Q&A]
Hi! I am trying to use casadi with C++ and windows, with the CMake file I can compile and execute the callback.cpp example and it works perfectly.
`But now I want to compile with this flags to use ipopt:
        {
            "name": "common",
            "hidden": false,
          "cacheVariables": {
            "WITH_BUILD_REQUIRED": "ON",
            "WITH_IPOPT": "ON",
            "WITH_OPENMP": "ON",
            "WITH_QPOASES": "ON",
            "WITH_LAPACK": "ON"
          }
        },
        
And it gives the following error:
Severity Code Description Project File Line Suppression State
Error 'external_projects/lib/casadi-tp-openblas.lib', needed by 'casadi_linsol_lapacklu.dll', missing and no known rule to make it. ninja 1	
Any idea about what can I do? I am trying to execute this script:
```
int main() {
    // // Optimization variables
    MX x = MX::sym("x", 2);

    ////// Objective
    MX f = x(0) * x(0) + x(1) * x(1);

    ////// Constraints
    MX g = x(0) + x(1) - 10;

    //// Create an NLP solver instance
    Function solver = nlpsol("solver", "ipopt", { {"x", x}, {"f", f}, {"g", g} });

    //// Generate C code for the NLP functions
    solver.generate_dependencies("nlp.c");

    //// Just-in-time compilation?
    bool jit = false;
    if (jit) {
       // Create a new NLP solver instance using just-in-time compilation
       solver = nlpsol("solver", "ipopt", "nlp.c");
    }
    else {
       // Compile the c-code
       int flag = system("gcc -fPIC -shared -O3 nlp.c -o nlp.so");
       casadi_assert(flag == 0, "Compilation failed");

    //    // Create a new NLP solver instance from the compiled code
       solver = nlpsol("solver", "ipopt", "nlp.so");
    }

    //// Bounds and initial guess
    std::map<std::string, DM> arg, res;
    arg["lbx"] = -DM::inf();
    arg["ubx"] = DM::inf();
    arg["lbg"] = 0;
    arg["ubg"] = 0;
    arg["x0"] = 0;

    //// Solve the NLP
    res = solver(arg);

    //// Print solution
    std::cout << "-----" << std::endl;
    std::cout << "objective at solution = " << res.at("f") << std::endl;
    std::cout << "primal solution = " << res.at("x") << std::endl;
    std::cout << "dual solution (x) = " << res.at("lam_x") << std::endl;
    std::cout << "dual solution (g) = " << res.at("lam_g") << std::endl;

    return 0;
}
```
Thanks a lot for the help

**Comment by jgillis** [comment]
Can you list your `cmake` output?
What happens if you set `WITH_BUILD_LACPACK` to ON as well?
